dnsapi/dns_pdns.sh

```diff
@@ -88,17 +88,22 @@ set_record() {
   _info "Adding record"
   root=$1
   full=$2
-  txtvalue=$3
+  new_challenge=$3
+
+  _pdns_rest "GET" "/api/v1/servers/$PDNS_ServerId/zones/$root"
+  _record_string=""
+  _build_record_string "$new_challenge"
+  _existing_challenges=$(echo "$response" | _normalizeJson | grep -Po "\"name\":\"$fulldomain\\K.*?}]" | grep -Po 'content\":\"\\"\K[^\\]*')
```

Comment:
do not use `grep -Po`,  please use our function `_egrep_o()` instead.

Reply:
Ok, I will have to adapt the regex because _egrep_o() seems not to support non greedy modifiers which is enabled by the -P flag. Working on it...